File: DAGF-AlgortimoProgramacaoDeComputadores-exercicios/cap7/calculadora.c
#include <stdio.h>
 
 void limpar_entrada(){
	char c;
	while((c=getchar()) != '\n' && c != EOF){}
 }
 
 void main() {
     float a,b;
     char op;
     
     printf("Digite dois números e a operação (+,-,*,/): ");
     printf("\nA = ");
     scanf("%f",&a);
     printf("\nB = ");
     scanf("%f",&b);
     printf("Qual a operação a ser realizada? ");
     limpar_entrada();
     scanf("%c",&op);
     
     switch (op){
         case '+' : printf("%.1f %c %.1f = %.1f\n", a,op,b,a+b); break;
         case '-' : printf("%.1f %c %.1f = %.1f\n", a,op,b,a-b); break;
         case '*' : printf("%.1f %c %.1f = %.1f\n", a,op,b,a*b); break;
         case '/' : if ( b == 0) {
                    printf("A operação não pode ser realizada");
                    }
                    else {
                        printf("%.1f %c %.1f = %.2f\n", a,op,b,a/b);
                    }
                    break;
         default : printf("Operação Inválida");
     } 
     
     printf("\n\n");
     
 }
                 

File: DAGF/src/com/company/walescko/cap07/Calculadora.java
package com.company.walescko.cap07;

import java.util.Scanner;

public class Calculadora {//Calculadora outra vez... em java...

    public static void main(String[] arg){

        Scanner scan = new Scanner(System.in);

        int numeroA, numeroB;
        char opcao;
        String escolha, saida ="";

        System.out.println("Exemplo Capítulo 07 - Estruturas Aninhadas - Calculadora");
        System.out.println("Digite dois numeros e a operação (+, -, * , /):");
        System.out.print("Numero A: ");
        numeroA = scan.nextInt();
        System.out.print("Numero B: ");
        numeroB = scan.nextInt();
        System.out.print("Escolha a operacao: ");
        escolha = scan.next();
        opcao = escolha.charAt(0);

        switch(opcao){
            case '+': saida += + numeroA + " + " + numeroB + " = " + (numeroA + numeroB); break;
            case '-': saida += + numeroA + " - " + numeroB + " = " + (numeroA - numeroB); break;
            case '*': saida += + numeroA + " * " + numeroB + " = " + (numeroA * numeroB); break;
            case '/': {
                if (numeroB == 0) {
                    saida = "A operacao nao pode ser realizada";
                } else {
                    saida += +numeroA + " / " + numeroB + " = " + (numeroA / numeroB);
                }
                break;
            }
            default: saida = "Operacao Invalida";
        }
        System.out.println(saida);
    }
}
